chalicelib/checks/wrangler_checks.py

```diff
@@ -2412,19 +2412,29 @@ def compare_record(existing_record, user, all_labs, all_grants, new=False):
                 updates[a_key] = a_val
         return updates
 
-    def find_lab(record, all_labs):
+    def find_lab(record, all_labs, all_grants):
         lab = ''
         all_lab_names = [i['display_title'] for i in all_labs]
         score = 0
         best = ''
+        log = []
         for disp in all_lab_names:
             s = fuzz.token_sort_ratio(record['OH Lab'], disp.split(',')[0])
             if s > score:
                 best = disp
                 score = s
         if score > 73:
             lab = [i['@id'] for i in all_labs if i['display_title'] == best][0]
-        return lab, score
+
+        if not lab:
+            oh_grant = record.get('OH Grant', '')
+            grant = [i for i in all_grants if i['name'].endswith(oh_grant)]
```

Comment:
So here is there was not a value in the OH Grant field then you will end up with a list of all the grants because you are looking for any award that ends with the empty string which is all of them.
i.e. if oh_grant = '' then if i['name'].endswith(oh_grant) will be true.
Is that what you intend? Because you just grab the first one for assigning the lab.